### ISSUE TITLE
Incorrectly inferred return type for `Psl\Iter\first(...)`

### ISSUE THREAD

**Issue by evang522** [php-standard-library/psalm-plugin#12]
**Describe the bug**
When provided an array with type `list<string>`, the output of Psl\Iter\first($array) is considered by the plugin's `FunctionReturnTypeProvider` to be never null, which is presumably not true. 

**To Reproduce**

Run Psalm with this plugin enabled on the following code:
```php
/** @var list<string> $series  **/
$series = ['val1', 'val2'];

$firstInSeries = \Psl\Iter\first($series);

\assert($firstInSeries !== null); // assert with any assertion module you like, it's a similar output for all.
```

Output: 

```txt
ERROR: RedundantConditionGivenDocblockType - ../../test.php:32:9 - Docblock-defined type string can never contain null (see https://psalm.dev/156)
\assert($firstInSeries !== null);
```

---

When duplicating the code of \Psl\Iter\first so that the plugin is not applied, this problem does not occur: 
```php
/**
 * @template T
 *
 * @param iterable<T> $iterable
 *
 * @return T|null
 */
function first(iterable $iterable)
{
    foreach ($iterable as $v) {
        return $v;
    }

    return null;
}



/** @var list<string> $series  **/
$series = ['val1', 'val2'];

$firstInSeries = first($series);

\assert($firstInSeries !== null);

```
Output:  No errors found!

**Expected behavior**
\Psl\Iter\first return type is considered to be possibly null as long as the array is not `non-empty-list<*>`

**Environment (please complete the following information):**
 - OS: fedora
 - PHP version: 8.1
 - Version: 2.2.1

Thanks a lot!

**Comment by azjezz** [comment]
this shouldn't really happen unless the type is known not to be empty ( https://github.com/php-standard-library/psalm-plugin/blob/main/src/EventHandler/Iter/First/FunctionReturnTypeProvider.php#L36-L42 ), i will look further into it 👍
